app/models/application_key.rb

```diff
@@ -23,7 +23,7 @@ class ApplicationKey < ApplicationRecord
   after_commit :push_webhook_key_destroyed, :on => :destroy
 
   after_commit :notify_after_create, :on => :create, :if => :should_notify?
-  after_commit :notify_after_destroy, :on => :destroy, :if => :should_notify?
+  after_commit :notify_after_destroy, :on => :destroy, if: proc { should_notify? && !destroyed_by_association }
```

Comment:
let's use `->` in here, instead of proc. better to stick with one way and not confuse juniors.

Reply:
@macejmic please use `suggestion` so I can commit from here 😎

Reply:
```suggestion
after_commit :notify_after_destroy, :on => :destroy, if: -> { should_notify? && !destroyed_by_association }
```